activities.el

```diff
@@ -801,6 +851,116 @@ activity's name is NAME."
                   "In the meantime, it's recommended to not use buffers of this major mode in an activity's layout; or you may simply ignore this error and use the other buffers in the activity.")
           (current-buffer)))))
 
+(defvar activities--age-spec
+  `((?Y "year"   "years"   ,(round (* 60 60 24 365.2425)))
+    (?M "month"  "months"  ,(round (* 60 60 24 30.436875)))
+    (?w "week"   "weeks"   ,(* 60 60 24 7))
+    (?d "day"    "days"    ,(* 60 60 24))
+    (?h "hour"   "hours"   ,(* 60 60))
+    (?m "minute" "minutes" 60)
+    (?s "second" "seconds" 1))
+  "Age specification.  See `magit--age-spec', which this duplicates.")
```

Comment:
See copyright question below.

Reply:
I think the right approach is to get something like this into time-date.el, copy it here temporarily, and once it shows up in `compat`, remove it.  I'll send in a patch.

Reply:
One issue is this similar existing function:

```elisp
(defvar seconds-to-string
  (list (list 1 "ms" 0.001)
        (list 100 "s" 1)
        (list (* 60 100) "m" 60.0)
        (list (* 3600 30) "h" 3600.0)
        (list (* 3600 24 400) "d" (* 3600.0 24.0))
        (list nil "y" (* 365.25 24 3600)))
  "Formatting used by the function `seconds-to-string'.")
;;;###autoload
(defun seconds-to-string (delay)
  ;; FIXME: There's a similar (tho fancier) function in mastodon.el!
  "Convert the time interval in seconds to a short string."
  (cond ((> 0 delay) (concat "-" (seconds-to-string (- delay))))
        ((= 0 delay) "0s")
        (t (let ((sts seconds-to-string) here)
             (while (and (car (setq here (pop sts)))
                         (<= (car here) delay)))
             (concat (format "%.2f" (/ delay (car (cddr here)))) (cadr here))))))
```

Reply:
See bug#71573.

Reply:
FWIW, I replied to that bug report linking to this code in my `ts` library: https://github.com/alphapapa/ts.el/blob/master/ts.el#L440-L491  Although it works a bit differently, I've used it extensively in some of my other packages, and profiling shows that it performs very well in terms of runtime and GC.  If any of it seems useful, I'll be glad to contribute it upstream as well (I'd be glad to contribute all of `ts`, in fact, but I haven't exactly proposed that yet).

Reply:
Thanks.  My guess is the attention to cutting v30 will mean this sits for a while.  I tried to adapt to the style of date-time but am not committed to any particular version. I would like to include weeks and months; to me that's the key elaboration of `magit--age`
(plus the modest resolution).  

How about we give it a week and if no movement, put in a version ourselves, with the aim to remove it once the patch is included in a release (+compat)?

Reply:
Some uptake on the patch adding a new `readable` option to `seconds-to-string`, but Eli just confirmed it wouldn't land until Emacs 31, so I propose we include a copy here along the same lines, with a note to replace it eventually.  Pushed changes along these lines.